Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp

```diff
@@ -36,34 +38,186 @@ namespace AZ::Internal
 
     static constexpr const char* ProductCacheDirectoryName = "Cache";
 
-    AZ::SettingsRegistryInterface::FixedValueString GetEngineMonikerForProject(
-        SettingsRegistryInterface& settingsRegistry, const AZ::IO::FixedMaxPath& projectJsonPath)
+    AZ::Outcome<void, AZStd::string> MergeSettingsFile(
+        AZ::SettingsRegistryInterface& settingsRegistry, const AZ::IO::FixedMaxPath& path, AZStd::string_view anchorKey)
     {
-        // projectPath needs to be an absolute path here.
+        if (!AZ::IO::SystemFile::Exists(path.c_str()))
+        {
+            return AZ::Failure(AZStd::string::format("The setreg file: '%s' does not exist and cannot be merged.", path.c_str()));
+        }
+        else if (!settingsRegistry.MergeSettingsFile(path.Native(), AZ::SettingsRegistryInterface::Format::JsonMergePatch, anchorKey))
+        {
+            return AZ::Failure(AZStd::string::format("Could not merge '%s' registry settings.", path.c_str()));
+        }
+        return AZ::Success();
+    }
```

Comment:
An Outcome is convertible to bool via it's operator bool.

How about we just updated the `SettingsRegistryImpl::MergeSettingsFile` and `SettingsRegistry::MergeSettingsFileInternal` function to return an `AZ::Outcome<void, AZStd::string>` instead and update all the AZ_Warning and AZ_Error calls within those functions to instead be sent to the error string that is returned?